mne_bids_pipeline/steps/preprocessing/_02_head_pos.py

```diff
@@ -148,16 +148,17 @@ def get_input_fnames_twa_head_pos(
     cfg: SimpleNamespace,
     subject: str,
     session: str | None,
-    run: str | None,
+    run: str | None = None,
     task: str | None,
 ) -> dict[str, BIDSPath | list[BIDSPath]]:
     """Get paths of files required by compute_twa_head_pos function."""
     in_files: dict[str, BIDSPath] = dict()
     # can't use `_get_run_path()` here because we don't loop over runs/tasks.
     # But any run will do, as long as the file exists:
-    run, _ = get_runs_tasks(
+    runs_tasks = get_runs_tasks(
         config=cfg, subject=subject, session=session, which=("runs",)
-    )[0]
+    )
+    run = next(filter(lambda run_task: run_task[1] == task, runs_tasks))[0]
```

Comment:
previously there was a possibility that `runs_tasks[0][0]` might be a run that wasn't present for the current `task`; this should correct that.